sdks/unity/AgonesSdk.cs

```diff
@@ -325,6 +345,29 @@ private void OnRequestCompleted(AsyncOperation _)
                 continuation = null;
             }
         }
+
+        /// <summary>
+        /// Custom UnityWebRequest http data handler
+        /// that fires a callback whenever it receives data
+        /// from the SDK.Watch() REST endpoint 
+        /// </summary>
+        private class GameServerHandler : DownloadHandlerScript
+        {
+            private WatchGameServerCallback callback;
+            public GameServerHandler(WatchGameServerCallback callback)
+            {
+                this.callback = callback;
+            }
+
+            protected override bool ReceiveData(byte[] data, int dataLength)
+            {
+                string json = Encoding.UTF8.GetString(data);
+                var dictionary = (Dictionary<string, object>) Json.Deserialize(json);

```

Comment:
can the data contain invalid json?

Reply:
If it did, it would be Exceptional.

:rofl: